daemon/src/taker_cfd.rs

```diff
@@ -60,8 +60,8 @@ pub struct Actor {
     send_to_maker: Address<send_to_socket::Actor<wire::TakerToMaker>>,
     monitor_actor: Address<monitor::Actor<Actor>>,
     setup_state: SetupState,
-    latest_announcement: Option<oracle::Announcement>,
-    _oracle_actor: Address<oracle::Actor<Actor, monitor::Actor<Actor>>>,
+    latest_announcements: Option<oracle::Announcements>,
```

Comment:
Can we make this a `HashMap` instead? Sounds like the better data structure if we are going to look-up by event ID.

Reply:
Just make sure to actively clear it on every update, otherwise we have a memory leak here.

Reply:
Went for keeping the array in the oracle and map + replace on every update in the other actors - ok?